npm/private/npm_import.bzl

```diff
@@ -472,17 +472,25 @@ def _impl_links(rctx):
     deps = {}
 
     for (dep_name, dep_version) in rctx.attr.deps.items():
-        if dep_version.startswith("/"):
-            store_package, store_version = utils.parse_pnpm_name(dep_version[1:])
+        if dep_version.startswith("link:") or dep_version.startswith("file:"):
+            dep_store_target = """"//{root_package}:{virtual_store_root}/{{}}/{package}/{version}".format(link_root_name)""".format(
+                root_package = rctx.attr.root_package,
+                package = dep_name,
+                version = "0.0.0",
+                virtual_store_root = utils.virtual_store_root,
+            )
         else:
-            store_package = dep_name
-            store_version = dep_version
-        dep_store_target_ref = """":{virtual_store_root}/{{}}/{package}/{version}/ref".format(link_root_name)""".format(
-            package = store_package,
-            version = store_version,
-            virtual_store_root = utils.virtual_store_root,
-        )
-        ref_deps[dep_store_target_ref] = ref_deps[dep_store_target_ref] + [dep_name] if dep_store_target_ref in ref_deps else [dep_name]
+            if dep_version.startswith("/"):
+                store_package, store_version = utils.parse_pnpm_name(dep_version[1:])
+            else:
+                store_package = dep_name
+                store_version = dep_version
+            dep_store_target = """":{virtual_store_root}/{{}}/{package}/{version}/ref".format(link_root_name)""".format(
```

Comment:
oh god the format() calls are nested... 🩸 👁️ 
and it starts with four double quotes

maybe using `":{virtual_store_root}/%s/{package}/{version}/ref" % link_root_name` syntax for the inner interpolation would help readability?

Reply:
not a bad idea, will save that for refactoring :)